index.ts

```diff
@@ -0,0 +1,37 @@
+import products from "./data/products.json";
+import { Cart, Product } from "./interfaces";
+
+const addProduct = (cart: Cart, product: Product) => {
+  cart[product.name] = { ...product };
+};
+
+const removeProduct = (cart: Cart, productToRemove: Product) => {
+  Object.keys(cart).filter(
```

Comment:
you need to edit the cart obj. 
check if the product is indeed inside the cart and remove using `delete`